app/routes.py

```diff
@@ -1 +1,167 @@
-from flask import Blueprint
\ No newline at end of file
+from flask import Blueprint, request, jsonify
+from app import db
+from app.models.task import Task
+from app.models.goal import Goal
+from app.helper_functions import validate_model, query_sort, filter_results, filter_and_sort, send_slack_message
+from datetime import datetime
+
+tasks_bp = Blueprint("tasks", __name__, url_prefix="/tasks")
+goals_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+@tasks_bp.route("", methods=["GET"])
+def get_all_tasks():    
+    tasks = filter_and_sort(Task)
+    
+    tasks_response = [task.to_json() for task in tasks]
+    
+    return jsonify(tasks_response), 200
+
+# Route to get task by task id
+@tasks_bp.route("/<task_id>", methods=["GET"])
+def get_task_by_id(task_id):
+    task = validate_model(Task, task_id)
+
+    return {"task": task.to_json()}, 200
+
+# Route to create new task
+@tasks_bp.route("", methods=["POST"])
+def create_task():
+    new_task = Task.from_json(request.get_json())
+    
+    db.session.add(new_task)
+    db.session.commit()
+
+    return {"task": new_task.to_json()}, 201
+
+@tasks_bp.route("/<task_id>", methods=["PUT"])
+def update_task(task_id):
+    task = validate_model(Task, task_id)
+    request_body = request.get_json()
+    
+    task.title = request_body["title"],
+    task.description = request_body["description"]
```

Comment:
Right now, if a user sends a request without the keys title or description your server would crash. There's a couple of ways to handle this, you could call the `validate_request` function before you access the keys in `request_body` or you could implement a `try/except` block.